packages/sdk/src/pen.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  Bip39,
+  EnglishMnemonic,
+  Secp256k1,
+  Sha256,
+  Sha512,
+  Slip10,
+  Slip10Curve,
+  Slip10RawIndex,
+} from "@iov/crypto";
+
+export type PrehashType = "sha256" | "sha512" | null;
+
+/**
+ * A pen is the most basic tool you can think of for signing. It works
+ * everywhere and can be used intuitively by everyone. However, it does not
+ * come with a great amount of features. End of semi suitable metaphor.
+ *
+ * This wraps a single keypair and allows for signing.
+ *
+ * Non-goals of this types are: multi account support, persistency, data migrations,
+ * obfuscation of sensitive data.
+ */
+export interface Pen {
+  readonly pubkey: Uint8Array;
+  readonly createSignature: (signBytes: Uint8Array, prehashType?: PrehashType) => Promise<Uint8Array>;
+}
+
+function prehash(bytes: Uint8Array, type: PrehashType): Uint8Array {
+  switch (type) {
+    case null:
+      return new Uint8Array([...bytes]);
+    case "sha256":
+      return new Sha256(bytes).digest();
+    case "sha512":
+      return new Sha512(bytes).digest();
+    default:
+      throw new Error("Unknown prehash type");
+  }
+}
+
+/**
+ * The Cosmoshub derivation path in the form `m/44'/118'/0'/0/a`
+ * with 0-based account index `a`.
+ */
+export function makeCosmoshubPath(a: number): readonly Slip10RawIndex[] {
+  return [
+    Slip10RawIndex.hardened(44),
+    Slip10RawIndex.hardened(118),
+    Slip10RawIndex.hardened(0),
+    Slip10RawIndex.normal(0),
+    Slip10RawIndex.normal(a),
+  ];
+}
+
+export class Secp256k1Pen implements Pen {
+  public static async fromMnemonic(
+    mnemonic: string,
```

Comment:
I like this interface, and for all single signer examples works well.

Default path with override is very nice